### PR TITLE
Dynamic listeners 

### PR DESCRIPTION
**Pull request by wtrocki**
## Motivation

Enable `client.registerOfflineEventListener` to dynamically register global listeners. 

ping @StephenCoady

### CODE REVIEW

**Comment by wtrocki** [comment]
I have messed up something with the git. Going to fix that.

**Comment by wtrocki** [comment]
Ready for review. Going to improve api docs.

**Comment by wtrocki** [comment]
@StephenCoady Applied fix for the problem that we have mentioned on call.

File: packages/sync/src/offline/OfflineMutationsHandler.ts [hunk 1]
```diff
@@ -18,14 +18,12 @@ export const logger = debug.default(MUTATION_QUEUE_LOGGER);
...
-  private store: OfflineStore;
   private mutationCacheUpdates?: CacheUpdates;
 
-  constructor(apolloClient: ApolloClient<NormalizedCacheObject>, clientConfig: DataSyncConfig) {
-    this.apolloClient = apolloClient;
+  constructor(private store: OfflineStore,
+              private apolloClient: ApolloClient<NormalizedCacheObject>,
+              clientConfig: DataSyncConfig) {
     this.mutationCacheUpdates = clientConfig.mutationCacheUpdates;
-    this.store = new OfflineStore(clientConfig);
```
Comment (StephenCoady) [inline packages/sync/src/offline/OfflineMutationsHandler.ts]: shouldn't this assign from store in the constructor now?

Comment (wtrocki) [reply packages/sync/src/offline/OfflineMutationsHandler.ts]: contructor arguments with `private` clasiffier will get instantly an private field of the class. Is that what you mean?

Comment (StephenCoady) [reply packages/sync/src/offline/OfflineMutationsHandler.ts]: it certainly is 🤦‍♂

**Pull request merged**